### PR TITLE
Testing cleanups

### PR DESCRIPTION
**Pull request by ehildenb**
cleanups to testing harness extracted from #187 PR.

### CODE REVIEW

**Comment by ehildenb** [comment]
Jenkins: test this please

**Comment by ehildenb** [comment]
Jenkins: test this please

**Comment by ehildenb** [comment]
@dwightguth ready for review

**Comment by dwightguth** [comment]
Jenkins: test mantis

File: Makefile [hunk 1]
```diff
@@ -223,6 +223,10 @@ bad_vm_tests= $(wildcard tests/ethereum-tests/VMTests/vmBlockInfoTest/blockhash*
               tests/ethereum-tests/VMTests/vmEnvironmentalInfo/env1.json \
               tests/ethereum-tests/VMTests/vmEnvironmentalInfo/extcodecopy0AddressTooBigRight.json \
               tests/ethereum-tests/VMTests/vmEnvironmentalInfo/ExtCodeSizeAddressInputTooBigRightMyAddress.json \
+              tests/ethereum-tests/VMTests/vmRandomTest/201503102037PYTHON.json \
+              tests/ethereum-tests/VMTests/vmRandomTest/201503102148PYTHON.json \
+              tests/ethereum-tests/VMTests/vmRandomTest/201503102300PYTHON.json \
+              tests/ethereum-tests/VMTests/vmRandomTest/201503110050PYTHON.json \
```
Comment (dwightguth) [inline Makefile]: test-all-vm needs to not test the bad_vm_tests.

**Comment by ehildenb** [comment]
@dwightguth I've excluded `bad_vm_tests` from target `test-all-vm`. I do intend to eventually completely remove the `VMTESTS` mode from the semantics, but I think it will have to wait for https://github.com/ethereum/tests/pull/444 (and I assume others which follow) to finish because currently we still need special behaviour for VMTests mode in `driver.md`.

Once all the "bad" VMTests are removed (that is, all the VMTests which interact with network state at all), we should be able to completely remove VMTests mode.

**Pull request merged**